pkg/unstruct/v1alpha2/unstruct_test.go

```diff
@@ -0,0 +1,106 @@
+package v1alpha2
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+const (
+	fakeK8sResource = `
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: icstcee
+  name: icstcee
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: icstcee
+  type: LoadBalancer
+`
+	fakeInvalidK8sResource = `
+apiVersion: v1
+kind: Service
+metadata
+  labels
+    app: icstcee
+  name: icstcee
+spec:
+  ports:
+  - port: 80
+    protocol: TCP
+    targetPort: 80
+  selector:
+    app: icstcee
+  type: LoadBalancer
+	`
+)
+
+func fakeUnstructObject(name string) *unstructured.Unstructured {
+	u := &unstructured.Unstructured{}
+	u.SetName(name)
+	return u
+}
+
+func TestUnstructForYAMLBuilder(t *testing.T) {
```

Comment:
Can you rename based on the target function i.e. the function that is being tested here.

Reply:
changed